verify/verify.go

```diff
@@ -472,12 +476,27 @@ type Options struct {
 	// Getter takes a URL and returns the body of its contents. By default uses http.Get and returns
 	// the body.
 	Getter trust.HTTPSGetter
+	// KDSClockSkewThreshold is the length of time permitted to wait for a certificate from KDS to
+	// become valid. The host and KDS servers' clocks may be skewed such that a VCEK certificate
+	// may have been "certified in the future".
+	KDSClockSkewThreshold time.Duration
+	// Now is the time at which to verify the validity of certificates. If unset, uses time.Now().
+	Now time.Time
 	// TrustedRoots specifies the ARK and ASK certificates to trust when checking the VCEK. If nil,
 	// then verification will fall back on embedded AMD-published root certificates.
 	// Maps the product name to an array of allowed roots.
 	TrustedRoots map[string][]*trust.AMDRootCerts
 }
 
+// DefaultOptions returns a useful default verification option setting
+func DefaultOptions() *Options {
+	return &Options{
+		Getter:                trust.DefaultHTTPSGetter(),
+		KDSClockSkewThreshold: 5 * time.Second,
```

Comment:
Just wandering how is this skew determined? I would expect this to be as small as possible.

Reply:
The failure we saw was within a second, and computers using the network time protocol shouldn't be off by much. Windows says a maximum skew should be 5 minutes https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/maximum-tolerance-for-computer-clock-synchronization

The Unix tolerance is 550 parts per million for the frequency offset. 100 PPM = 8.64 seconds / day, which is about 48 seconds per day of drift.

So maybe split the difference and this should be a minute? The 5 seconds was a reaction to the 1 second error and trying to keep to the mechanical engineering principle of "design structural tolerance to 5 times the expected maximum capacity."

Let me know which value you think is fair for a default.

Reply:
5 minutes sound okay for now as discussed offline.